design-proposals/sdn-ipam-secondary-nets.md

```diff
@@ -0,0 +1,482 @@
+not impacted.
+- As a VM user, I want to start Virtual Machines (currently stopped) with the
+original VM network configuration, so the IP-aware applications running in the
+VM are not impacted.
+- As a network admin, I want to decide which networks can be subject to having
+persistent IP allocations surviving VM migration, and VM stop.
+
+## Goals
+- Manage KubeVirt Virtual Machines IP addresses from within the SDN solution
+provided by OVN-Kubernetes.
+- IPv4 range can be defined on OVN Kubernetes secondary network
+- VMs are able to get a IP from this subnet
+- The IP sticks to a VM during live-migration
+- The IP sticks to a VM even if it is temporarily shut down
+
+## Non-goals
+- this design proposal's scope is limited to secondary networks - i.e. the
+default cluster network's IP allocation should still follow the pod lifecycle.
+- seamless live migration. Some packets will be dropped during live migration;
+this feature is tracked in a follow up epic.
+
+## Repos
+- [CRD](https://github.com/k8snetworkplumbingwg/ipamclaims)
+- [CNI plugin](https://github.com/ovn-org/ovn-kubernetes)
+- [KubeVirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+This feature is essentially about tying the IPAM allocation to a VM custom
+resource, instead of a pod.
+
+This means we want the IP allocations to remain when the VM is stopped (thus the
+corresponding VMI and pod are deleted), but for them to be released to the IP
+pools when the VM CR is deleted.
+
+Given OVN-Kubernetes (the CNI plugin where this feature will be implemented)
+operates at pod level (i.e. it does **not** know what a KubeVirt VM is), and its
+source of truth is in essence the live pods on the cluster, we need to have
+something in the data model representing this existing allocation when the pod
+is deleted (the stopped VM scenario). If we don't, those IP addresses would be
+allocatable for other workloads while the VM is stopped.
+
+Thus, we propose to introduce a new CRD to the k8snetworkplumbingwg, and make it
+part of the Kubernetes multi-network defacto standard in this update proposal.
+The `IPAMClaim` CRD was added to the Kubernetes multi-networking de-facto
+[standard](https://github.com/k8snetworkplumbingwg/multi-net-spec/blob/master/v1.3/%5Bv1.3%5D%20Kubernetes%20Network%20Custom%20Resource%20Definition%20De-facto%20Standard.pdf)
+version 1.3. The sections describing the CRD and how to use them are in sections
+4.1.2.1.11 (network selection element update) and 8 (IPAMClaim CRD).
+
+The IPAMClaim CRD is available in
+[this repo](https://github.com/k8snetworkplumbingwg/ipamclaims).
+
+The CNI plugin must be adapted to compute its IP pool not only from the live
+pods in the cluster, but also from these CRs.
+
+### Configuring the feature
+We envision this feature to be configurable per network, meaning the network
+admin should enable the feature by enabling the `allowPersistentIPs` flag in the
```

Comment:
we're actually discussing right now ignore vs error. 

I think it should error, because it's pretty much clear we won't give the user what they want. 

Thus failing seems more "honest".